src/main/java/org/example/requestsAndConnections/GameController.java

```diff
@@ -0,0 +1,59 @@
+package org.example.requestsAndConnections;
+
+import org.example.dto.GameStateDTO;
+import org.example.dto.StepForMoveDTO;
+import org.example.game.Game;
+import org.example.game.Sessions;
+import org.example.myExeptions.MyCustomExceptions;
+
+import java.util.UUID;
+
+public class GameController {
+    static Sessions sessions = new Sessions();
+
+    public UUID newSession() {
+        UUID sessionId = UUID.randomUUID();
+        sessions.addSession(sessionId);
+        return sessionId;
+    }
+
+    public GameStateDTO newGame(UUID sessionId) throws MyCustomExceptions {
+        if (!sessions.containsSession(sessionId)) {
+            throw new MyCustomExceptions("Invalid session id");
+        }
+        return createAndPutNewGameToSessions(sessionId).getGameState();
+    }
+
+    public GameStateDTO getState(UUID sessionId) throws MyCustomExceptions {
+        if (!sessions.containsSession(sessionId)) {
+            throw new MyCustomExceptions("Invalid session id");
+        }
+        if (sessions.hasGames(sessionId)) {
+            Game game = sessions.getLastAddedGame(sessionId);
+            return game.getGameState();
+        } else return newGame(sessionId);
+    }
+
+    public GameStateDTO move(StepForMoveDTO dataForStep) throws MyCustomExceptions {
+        if (!sessions.containsSession(dataForStep.getSession_id())) {
+            throw new MyCustomExceptions("Invalid session id");
+        }
+        if (!sessions.hasGames(dataForStep.getSession_id())) {
+
+            return createAndPutNewGameToSessions(dataForStep.getSession_id()).move(dataForStep);
```

Comment:
Again, you are trying to make a move, but in the end - you are creating a game. It can save you some time - but in the long run, it will lead to misunderstanding of behavior by you and other developers. Move is a move. If it can not be done - just throw and error.